server/controllers/refer/referralsController.ts

```diff
@@ -0,0 +1,41 @@
+import type { Request, Response, TypedRequestHandler } from 'express'
+import createError from 'http-errors'
+
+import type { CourseService, OrganisationService } from '../../services'
+import { courseUtils, typeUtils } from '../../utils'
+
+export default class ReferralsController {
+  constructor(
+    private readonly courseService: CourseService,
+    private readonly organisationService: OrganisationService,
+  ) {}
+
+  start(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      typeUtils.assertHasUser(req)
+
+      const course = await this.courseService.getCourse(req.user.token, req.params.courseId)
+      const courseOffering = await this.courseService.getOffering(
```

Comment:
Do we need to account for the possibility here of an offering not being found, like we do for Organisations?

Reply:
We don't in the course offerings controller's show action (or either of the course controller's actions or the offering one for courses). I believe our thinking was that organisations are external so we can't guarantee what comes back from the prison register API, whereas we have more confidence in our own data/API. Of course someone could play with the IDs in the URL but I think we previously decided only to deal with the clients that touch external data. Not sure what happens if we try to access a page using a non-existent offering or course ID